live-flowchart/sw-util.js

```diff
@@ -0,0 +1,178 @@
+    Logger.info('Please enable and check the browser logs for the oninstall, onactivate, and onfetch events');
+    Logger.debug('More on about:serviceworkers (Firefox) or chrome://serviceworker-internals/');
+
+    return true;
+  }
+
+  Logger.warn('No, there is no service worker controlling this app');
+  return false;
+};
+
+/*
+ * @method registerServiceWorker
+ * Register a service worker
+ * @param {String} path The script URL of the service worker to be registered
+ * @param {String} scope (optional) The scope of the service worker to be registered
+ * @returns {Promise}
+ */
+SWUtil.prototype.registerServiceWorker = function registerServiceWorker(scriptURL, scope) {
+  var swRegisterSecondParam = {};
+
+  Logger.log('\nSWUtil.registerServiceWorker()');
+
+  Logger.info('Register service worker with serviceWorker.register()');
+
+  Logger.debug('navigator.serviceWorker.register(scriptURL, scope)');
+  Logger.debug('scriptURL: ' + scriptURL);
+
+  // add optional parameter 'scope' if specified
+  if (scope) {
+    swRegisterSecondParam.scope = scope;
+    Logger.debug('options.scope: ' + swRegisterSecondParam.scope);
+  }
+
+  // return a promise
+  return new Promise(
+    function then(resolve, reject) {
+      navigator.serviceWorker.register(
+        scriptURL,
+        swRegisterSecondParam
+      ).then(
+        function registrationSuccess(swRegistration) {
+          Logger.debug('registrationSuccess(swRegistration)');
+          Logger.debug(swRegistration);
+
+          if (swRegistration.active) {
+            Logger.debug('Registering the active service worker again: ');
+            Logger.debug(swRegistration.active.scriptURL + ' <= swRegistration.active.scriptURL');
+          }
+
+          if (swRegistration.installing) {
+            Logger.debug('Registering the following service worker for the first time: ');
+            Logger.debug(swRegistration.installing.scriptURL + ' <= swRegistration.installing.scriptURL');
+          }
+
+          if (swRegistration.scope) {
+            Logger.debug('with scope: ' + swRegistration.scope + ' <= swRegistration.scope');
+          }
+
+          Logger.info('Service worker registered');
+          Logger.info('Please enable and check the browser logs for the oninstall, onactivate, and onfetch events');
+          Logger.info('SW is in control, once document is reloaded');
+          Logger.debug('More on about:serviceworkers (Firefox) or chrome://serviceworker-internals/');
+
+          resolve();
+        },
+        function registrationError(why) {
+          Logger.debug('registrationError(why)');
+          Logger.error(why);
+
+          reject();
+        }
+      );
+    }
+  );
+};
+
+/*
+ * @method unregisterServiceWorker
+ * Unregister the active service worker
+ * @returns {Promise}
+ */
+SWUtil.prototype.unregisterServiceWorker = function unregisterServiceWorker() {
+  Logger.log('\nSWUtil.unregisterServiceWorker()');
```

Comment:
Yep, both in the browser log and in the HTML log. I can do the same with a `Logger.log('\n')` or a `Logger.log('---')` before instead, or alternatively remove the new lines. Does it break the doc generation?